pkg/types/types.go

```diff
@@ -84,28 +134,46 @@ type ResourcePool interface {
 	GetMounts(deviceIDs []string) []*pluginapi.Mount
 }
 
-// PciNetDevice provides an interface to get device specific information
-type PciNetDevice interface {
-	GetPFName() string
+// DeviceProvider provides interface for device discovery
+type DeviceProvider interface {
```

Comment:
A device provider could report the device type it handles instead of having the device type  to provider knowledge in sriovdp manager

Reply:
I'm not following you on this. Can you please explain little bit more?